components/collector/tests/source_collectors/dependency_track/test_dependencies.py

```diff
@@ -92,3 +92,9 @@ async def test_filter_by_project_name_and_version(self):
         self.set_source_parameter("project_versions", ["1.3", "1.4"])
         response = await self.collect(get_request_json_side_effect=[self.projects(), self.dependencies("1.0")])
         self.assert_measurement(response, value="1", entities=self.entities("1.0", "up-to-date"))
+
+    async def test_filter_by_project_version_when_project_has_no_version(self):
+        """Test filtering projects by version."""
+        self.set_source_parameter("project_versions", ["1.2", "1.3"])
+        response = await self.collect(get_request_json_return_value=self.projects(version=""))
+        self.assert_measurement(response, value="0", entities=[])
```

Comment:
Shouldn't we have at least one value returned so we know it can actually match something?

Reply:
Added a test that matches the version number.